### PR TITLE
Collision Improvements Part X

### PR DESCRIPTION
**Pull request by rgarmsen2295**
Adds BB updates on orientation and rotation change (only doing orientation lead to a tight bound at start but a crumy bound if you turned 90 degrees - now it's only suboptimal when moving diagonally, which is the thing @nurgan has described in the past).

Our previous discussion and proposed fixes did not completely solve the corner issue due to the fact that the player BB can be colliding with the X-normal of a hedge and the Y-normal simultaneously on the same or different hedges.  This is hard to explain in words, but with the BB debug spheres its pretty easy to see in-game.

Also, due to updating on rotation now, there's the problem of sliding diagonally clipping early as the BB is now axis-aligned (so less of a bug and more of a tradeoff).  And there is a degenerate case where if you drive parallel to a hedge and then turn 90 degrees the car's BB becomes merged with the hedge...not sure if there's a solution for that one without more advanced rotation collision resolution.  Possible, but you have to do it on purpose so not sure if worth effort.

tl;dr Improvements were made, but they aren't as impressive as hoped.  One thing that does reduce the corner issue is adjusting the epsilon in the reflect logic of the BB.  However, this also makes the sliding effect much less slide-y, so it's something I'll need to play around with a bit more.

I'm mostly making this PR for discussion, so let me know thoughts.  I've been "staring into the sun" for a ton of time now (even though the number of lines changed don't show it lol) so wanna continue to solicit feedback

### CODE REVIEW

**Comment by nurgan** [comment]
It feels like most of it are just necessary trade-offs we have to pay for our more or less simple collision detection (aka just using the BB, and on detected collision of BBs not go down to actual Geometry or more but better fitting BBs). Also we shouldn't go more complex for the collision detection, but actual games do, and I think thats a lot of why it looks so much better, idk how far we can get with the simple BB solution.
Only 2 things that come spontaneous to my mind:
- trying a epsilon for collision detection that is dependent on the size of the objects
- using a bounding sphere for the car (but stick with BB for the rest)

Code looks good.

**Review by nurgan — APPROVED** [review]

**Comment by rgarmsen2295** [comment]
I would agree on the trade-off part, hard to get everything we want without a lot of work.

Agree on being able to vary the epsilon; I'm going to play with it and add the ability to change from the default 1.5 before I merge this.

Yes on spheres for player, that's something I'd like to get it before we finish 100% (but that will be done after this - will take some work (not a ton, but extra) to get everything to support AABB and spheres)

**Comment by joncatanio** [comment]
I'm good with this, it does seem like we are spending a lot of effort on BB logic and work. I think we've each had a hand in it at this point. I think the focus right now would be to beautify the game and get it looking really clean. We will have a lot of time between now and 100% to just get together one day and try to optimize the bounding structures. For now lets try to go broad on development before we focus super hard on BB stuff again. Cuz right now it is playable and we have stopped the going through walls bug for the most part. Users can understand how to go in reverse, they can't figure out how to get unstuck. So as long as the player isn't locked in place then we should be fine for now. We can revisit soon.

**Comment by joncatanio** [comment]
I'm going to merge this I want a clean master

**Review by joncatanio — APPROVED** [review]
